packages/astro/src/@types/astro.ts

```diff
@@ -92,6 +92,10 @@ export interface AstroGlobal extends AstroGlobalPartial {
 	 * [Astro reference](https://docs.astro.build/en/reference/api-reference/#astrocanonicalurl)
 	 */
 	canonicalURL: URL;
+	/** The address (usually IP address) of the user. Used with SSR only.
+	 * 
+	 */
+	clientAddress: string;
```

Comment:
Not to overcomplicate things, but edge platforms provide more granular connection info than just the IP address. Rather than supporting the lowest common denominator, can we future-proof this API by exposing `Astro.connection` and integrations can set things like `Astro.connection.address`?

For example, Deno has [`ConnInfo`](https://doc.deno.land/https://deno.land/std@0.148.0/http/server.ts/~/ConnInfo) with both `localAddr` and `remoteAddr`.

Reply:
Yeah, that would make some since. The `clientAddress` name was taken from SvelteKit; Remix has a 3rd party library that uses that name too.

I'm open to the suggested change. Waiting to here what other people in core think.